### PR TITLE
Update arch diagram

### LINKED ISSUE #5376: Architecture is outdated - direct istio dependency
The [Kiali architecture section and diagram](https://kiali.io/docs/architecture/architecture/#kiali-architecture) both suggest that istio/istiod are indirect dependencies but now Kiali does directly rely on and communicate with istio/istiod to gather information about workloads and services.

### PR DESCRIPTION
**Pull request by nrfox**
Couldn't find the old image anywhere to update so I added a new image that shows that istiod is a required dependency but also adds a note in the description that it can be disabled. Also changes the Cluster API --> Kube API because IMO Cluster API is confusing.

Fixes kiali/kiali#5376

### CODE REVIEW

File: content/en/docs/Architecture/architecture.md [hunk 1]
```diff
@@ -49,8 +49,8 @@ It provides and controls the service mesh. Kiali and Istio are installed
 separately.
 
 Kiali needs to retrieve Istio data and configurations, which are exposed
-through Prometheus and the cluster API. This is the reason the diagram shows a
-dashed line: to denote an indirect dependency.
+through Prometheus, the Kubernetes API, and istiod. For environments where
```
Comment (leandroberetta) [inline content/en/docs/Architecture/architecture.md:52]: ultra minor detail but the comma before "and istiod" should be ommited right?

**Review by leandroberetta — COMMENTED** [review]
Just a ultra minor comment, but looks good.

File: content/en/docs/Architecture/architecture.md (hunk 1 above)
Comment (nrfox) [reply content/en/docs/Architecture/architecture.md:52]: The [oxford comma](https://en.wikipedia.org/wiki/Serial_comma) is a contentious subject actually. I've always put one on the end of a list but I'm certainly no authority on the subject. I'm happy to remove it and break with my tradition.

Comment (nrfox) [reply content/en/docs/Architecture/architecture.md:52]: It looks like other places on the same page in the docs use it as well.
> Kiali queries the Kubernetes API to retrieve, for example, definitions for
> namespaces, services, deployments, pods, and other entities. 
> ...
> 
> The Kubernetes API is also queried to retrieve Istio configurations like virtual
> services, destination rules, route rules, gateways, and quotas.

How do you feel about keeping it for consistency's sake?

Comment (leandroberetta) [reply content/en/docs/Architecture/architecture.md:52]: Feels ok!

**Review by leandroberetta — APPROVED** [review]
LG!

**Pull request merged**